app/controllers/catalog_controller.rb

```diff
@@ -9,5 +9,16 @@ class CatalogController < ApplicationController
       :rows => 20 
     }
   end
+  
+  def index
+    super
+    Rails.cache.read_multi(@document_list.map(&:ci_id)).tap do |ci_ids_to_proxy_urls|
+      @ids_to_cached_proxy = Hash[
+        @document_list.map do |document|
+          [document.id, ci_ids_to_proxy_urls[[document.ci_id]]]
+        end
+      ]
+    end
+  end
```

Comment:
@mccalluc when calling `super` here you are executing [this method](https://github.com/projectblacklight/blacklight/blob/v5.11.1/lib/blacklight/catalog.rb#L35-L49), which includes the `respond_to` method, which is responsible for doing the rendering. I'm wary of upsetting Rails magic if the return value of `#index` is altered... but this could just be a superstition on my part.